edge/stack/stack.go

```diff
@@ -512,8 +514,8 @@ func (manager *StackManager) buildDeployerParams(stackData client.EdgeStackData,
 
 	switch manager.engineType {
 	case EngineTypeDockerStandalone, EngineTypeDockerSwarm:
-		if len(stackData.RegistryCredentials) > 0 && strings.HasPrefix(stackData.Name, "edge-update-schedule") {
-			yml := yaml.NewDockerComposeYAML(fileContent, stackData.RegistryCredentials)
+		if (len(stackData.RegistryCredentials) > 0 || manager.awsConfig != nil) && strings.HasPrefix(stackData.Name, "edge-update-schedule") {
```

Comment:
tech debt: change the last condition from `strings.HasPrefix(stackData.Name, "edge-update-schedule")` to `stackData.AddCredentials`

Reply:
Can you please explain this more?
This registry credential injection here is only for remote update from private registry

Reply:
it's very specific. what if we change the name of the stack? this should be an edge stack property and not some magic string check